### ISSUE TITLE
Select subset of the grid

### ISSUE THREAD

**Issue by markkvdb** [open-meteo/open-meteo#638]
It would be great if there was a possibility to be able to select a subset of the total grid that models offer. Many use-cases don't require to have data for the entire world but worldwide models like GFS are still valuable (especially for long-term forecasts). However, one model run of GFS25 is multiple GBs in size, since it has worldwide coverage.

## Side note

Selecting which variables to keep and which models to take are application-wide settings. Currently, the `sync` and `download` functionality each require the user to configure what data they would like to serve on their server. Would it make sense to see if these application-wide settings could be shared across the downloader and syncer (and more parts if needed)?

**Comment by markkvdb** [comment]
Maybe the `download` command was not the one I meant but the original GRIB files are at some point transformed into the optimised `.om` files. It could be useful to have the option to have application-wide settings for what variables, grid and models you want to keep. The operations that parses the original files could share the same "filters" (variables, grid, models) as the sync operation.

Speaking of this original GRIB -> `.om` operation. It could be great if there was a command to parse a GRIB of any of the supported models. This is especially useful for parsing historical data that might not be available through the APIs of the weather services.

**Comment by patrick-zippenfenig** [comment]
I do not see a simple approach to parse GRIB files to `.om` files. While downloading, the code "knows" which model run, variable, units and grid systems it is working on. Some information can be found in GRIB files, but most of the logic is hard-coded. Meta attributes inside GRIB files can even be wrong. E.g. accumulation time-ranges of precipitation or whether a solar flux parameter is instantaneous or backwards averaged.

`.om` files are also time-series databases and the conversion process needs to transpose spatial to temporal data and interpolate missing values (Some models switch from 1-hourly to 3-hourly after a couple of days of forecast). It is not just a 1:1 version.

Over the years, I standardised download and conversion steps, but it is still far from a generic approach for all downloaders. If you briefly check the code for various downloaders, you will see many specific steps that apply to only one provider. 

> It could be useful to have the option to have application-wide settings for what variables, grid and models you want to keep. The operations that parses the original files could share the same "filters" (variables, grid, models) as the sync operation.

`.env` files are supported and I could further explore how to use them for such filters. In terms so parameter validation, using CLI arguments like `--only-variables temperature_2m` has the advantage of been more explicit and the underlaying framework takes care of parsing them. A combination of `.env` attributes and CLI arguments could work.

**Comment by markkvdb** [comment]
I see your point about the metadata. I noticed that as well when reading the code for the downloaders. The processing steps to convert into `.om` files makes sense indeed. Still, in the downloader process a file is downloaded *and then processed*. So, I'd say that it should be possible to replace the initial real download step with providing a file that you already have. Of course, this implies that the necessary meta data needs to be provided so that it makes sense.

The reason I'm asking is that I'd really like to import historical forecast data for data science purposes. I need to be able train my models on sufficient data (often going back years). I understand that this specific historical file import is not a high priority of the project right now, so I'd like to try if I can make it work. Could you maybe point me in the right direction to create such a historical importer?

**Comment by patrick-zippenfenig** [comment]
It really depends on which model. You mentioned MeteoFrance in https://github.com/open-meteo/open-data/issues/3, but which kind of model/domain and how are they structured? For the old SP1 files, you could reverse https://github.com/open-meteo/open-meteo/pull/561. Otherwise you need a loop over each run, read all messages from grib, convert them to a "temporary" `.om` spatial file and update each run into a time-series

**Comment by markkvdb** [comment]
I have access to the `SP<n>` and `HP<n>` files for the AROME HD model and ARPEGE EUROPE. Thanks for pointing to the old parser. I will see if I can find some time this week to see if I can make it work.

Just for my understanding. What do you mean with "update each run into a time-series"? I guess that these chunks need to be merged together for better performance. Is there some automatic way to handle this or was it always specific for each model (and files)?

**Comment by markkvdb** [comment]
Ah sorry, yes I have each run stored separately! I wouldn't want to convert my entire database in one go.

**Comment by markkvdb** [comment]
I tried to use the historical forecast API and it's very fast compared to the solution I'm using right now. Especially impressive when I take variables that are derived and require quite some on-the-fly computation. Bravo! Now, I'm quite sure that I want to move my old architecture to open meteo and find a way to import all my historical meteo france files...

**Comment by patrick-zippenfenig** [comment]
Sound great! Actually the performance of the `historical forecast API` will be improved a lot in the next weeks. How large are your archives? I still have the old ARPEGE WORLD 0.4° model archived, but I am not sure if it makes sense anymore as it was replaced with the 0.25° version.

**Comment by patrick-zippenfenig** [comment]
Yeah that would be interesting. Could you check for any license constraints? I think that it should not be an issue as similar data is now available as open-data.
